test/e2e/specs/onboarding/heroflow__write.ts

```diff
@@ -0,0 +1,136 @@
+/**
+ * @group calypso-release
+ */
+
+import {
+	DataHelper,
+	StartSiteFlow,
+	RestAPIClient,
+	SecretsManager,
+	NewSiteResponse,
+	TestAccount,
+} from '@automattic/calypso-e2e';
+import { Page, Browser } from 'playwright';
+import { apiDeleteSite } from '../shared';
+
+declare const browser: Browser;
+
+describe( DataHelper.createSuiteTitle( 'Onboarding heroflow: write' ), function () {
+	const blogName = DataHelper.getBlogName();
+	const tagline = '${ blogName } tagline';
+
+	let siteCreatedFlag: boolean;
+	let newSiteDetails: NewSiteResponse;
+	let restAPIClient: RestAPIClient;
+	let page: Page;
+
+	beforeAll( async function () {
+		// Set up the test site programmatically against simpleSiteFreePlanUser.
+		const credentials = SecretsManager.secrets.testAccounts.simpleSiteFreePlanUser;
+
+		restAPIClient = new RestAPIClient( credentials );
+		console.info( 'Creating a new test site.' );
+		newSiteDetails = await restAPIClient.createSite( {
+			name: blogName,
+			title: blogName,
+		} );
+		console.info( `New site created: ${ newSiteDetails.blog_details.url }` );
+		siteCreatedFlag = true;
+
+		// Launch browser.
+		page = await browser.newPage();
+
+		// Authenticate as simpleSiteFreePlanUser.
+		const testAccount = new TestAccount( 'simpleSiteFreePlanUser' );
+		await testAccount.authenticate( page );
+	} );
+
+	describe( 'Onboarding flow', function () {
+		let startSiteFlow: StartSiteFlow;
+
+		it( 'Navigate to setup > goals', async function () {
+			await page.goto(
+				DataHelper.getCalypsoURL(
+					`/setup/goals?siteSlug=${ newSiteDetails.blog_details.site_slug }`
+				)
+			);
+		} );
+
+		it( 'Skip goals step', async function () {
+			startSiteFlow = new StartSiteFlow( page );
+
+			await page.waitForLoadState( 'networkidle' );
```

Comment:
**suggestion**: if you are sticking with the `getCurrentStep` pattern, this `networkidle` call is unnecessary as the `getCurrentStep` method [already contains such wait](https://github.com/Automattic/wp-calypso/blob/a25293cee3a8135e3f9d604dcfc3d2b10327fd45/packages/calypso-e2e/src/lib/flows/start-site-flow.ts#L65).

Reply:
Personally though, I'm not a fan of the `getCurrentStep` structure as I feel an E2E test should take the user down a list of predefined expectations instead of allowing a way out with `if`s.